lib/CryptoNoteCore/Currency.cpp

```diff
@@ -152,29 +152,27 @@ bool Currency::getBlockReward(
 
     size_t blockGrantedFullRewardZone = blockGrantedFullRewardZoneByBlockVersion(blockMajorVersion);
     medianSize = std::max(medianSize, blockGrantedFullRewardZone);
-    if (currentBlockSize > UINT64_C(2) * medianSize) {
+    if (currentBlockSize > medianSize * UINT64_C(2)) {
         logger(TRACE)
             << "Block cumulative size is too big: " << currentBlockSize
-            << ", expected less than " << medianSize * 2;
+            << ", expected less than " << medianSize * UINT64_C(2);
         return false;
     }
 
     uint64_t penalizedBaseReward = getPenalizedAmount(baseReward, medianSize, currentBlockSize);
-    uint64_t penalizedFee = blockMajorVersion >= BLOCK_MAJOR_VERSION_2
-                            ? getPenalizedAmount(fee, medianSize, currentBlockSize)
-                            : fee;
-    if (cryptonoteCoinVersion() == 1) {
+    uint64_t penalizedFee = fee;
+    if (blockMajorVersion >= BLOCK_MAJOR_VERSION_2 || cryptonoteCoinVersion() == 1) {
         penalizedFee = getPenalizedAmount(fee, medianSize, currentBlockSize);
     }
 
-    uint64_t consistency = 1;
+    double consistency = 1.0;
     if (difficultyTarget() != 0) {
         // blockTarget is (Timestamp of New Block - Timestamp of Previous Block)
         consistency = blockTarget / difficultyTarget();
 
         // consistency range is 0..2
-        consistency = std::max<uint64_t>(consistency, 0);
-        consistency = std::min<uint64_t>(consistency, 2);
+        consistency = std::max<uint64_t>(consistency, 0.0);
```

Comment:
Change these to double too, and make casts:
```
consistency = std::max<double>(consistency, 0.0);
consistency = std::min<double>(consistency, 2.0);
```

Reply:
Fixed in the latest commit (`7ffb667`).